tests/lib/prepare.sh

```diff
@@ -74,7 +101,7 @@ prepare_classic() {
 
         systemctl stop snapd.service snapd.socket
 
-        update_core_snap_with_snap_exec_snapctl
+        update_core_snap_for_classic_reexec
```

Comment:
And we could only call `update_core_snap_for_classic_reexec` only when `MODIFY_CORE_SNAP_FOR_REXEC` is set to 1.